src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -236,8 +363,180 @@ export default {
         this.openmct.time.off('bounds', this.boundsChange);
         this.openmct.time.off('timeSystem', this.timeSystemChange);
         this.openmct.time.off('clock', this.clockChange);
+
+        // unsubscribe from related telemetry
+        if (this.hasRelatedTelemetry) {
+            for (let key of this.relatedTelemetry.keys) {
+                if (this.relatedTelemetry[key].unsubscribe) {
+                    this.relatedTelemetry[key].unsubscribe();
+                }
+            }
+        }
+
+        clearInterval(this.pollResizeImageContainerID);
     },
     methods: {
+        async initializeRelatedTelemetry() {
```

Comment:
This function is pretty long, could we break it up into multiple functions that resolve realtime and historical related telemetry separately?

Reply:
Done! Moved out to a class.